pkg/telemetry/telemetry.go

```diff
@@ -224,6 +224,14 @@ func (t *TelemetryClient) TrackMetric(metricname string, value float64, properti
 
 func (t *TelemetryClient) TrackTrace(name string, severity contracts.SeverityLevel, properties map[string]string) {
 	trace := appinsights.NewTraceTelemetry(name, severity)
+
+	kernelVersion, err := kernelVersionFunc(context.Background())
+	if err == nil {
+		t.trackWarning(err, "failed to get kernel version for trace")
```

Comment:
won't this cause infinite recursion in the error case?

`t.TrackTrace -> t.trackWarning -> t.TrackTrace`